pkg/httpsproxy/proxylet/proxylet.go

```diff
@@ -43,6 +45,30 @@ func getLocationIDFromEnv() string {
 	return value
 }
 
+func RunProxylet(test bool) {
+	logLevel := httpproxy.GetEnvWithDefaults("LOG_LEVEL", "debug")
+	level, levelerr := log.ParseLevel(logLevel)
+	if levelerr != nil {
+		log.Infof("No valid log level from ENV, defaulting to debug level was: %s", level)
+		level = log.DebugLevel
+	}
+	log.SetLevel(level)
+
+	proxyletObject, err := NewProxylet()
+	if err != nil {
+		log.WithError(err).Fatal("Failed to create proxylet object")
+	}
+	proxyletObject.RunHttpProxylet()
+
+	if test {
+		quit := make(chan os.Signal)
+		testing.NotifyOnAppExitMessageGeneric(proxyletObject.natsClient, quit)
+		<- quit
+	} else {
+		runtime.Goexit()
+	}
+}
```

Comment:
This is the block of code that I mentioned above that would've been duplicated. The only real changes are the `if test { ... } else { ... }` part.